solver/exporter.go

```diff
@@ -198,13 +204,18 @@ func (e *exporter) ExportTo(ctx context.Context, t CacheExporterTarget, opt Cach
 			}
 		}
 
+		e.k.mu.RLock()
```

Comment:
This is the message I am getting regarding this change:
```
WARNING: DATA RACE
Write at 0x00c000b457d0 by goroutine 32771:
  runtime.mapassign_fast64()
      /usr/lib/go/src/runtime/map_fast64.go:93 +0x0
  github.com/moby/buildkit/solver.(*cacheManager).getID()
      /home/denis/git/buildkit/solver/cachemanager.go:283 +0x114
  github.com/moby/buildkit/solver.(*cacheManager).getIDFromDeps()
      /home/denis/git/buildkit/solver/cachemanager.go:324 +0x3ab
  github.com/moby/buildkit/solver.(*cacheManager).getID()
      /home/denis/git/buildkit/solver/cachemanager.go:287 +0x1a4
  github.com/moby/buildkit/solver.(*cacheManager).getIDFromDeps()
      /home/denis/git/buildkit/solver/cachemanager.go:324 +0x3ab
  github.com/moby/buildkit/solver.(*cacheManager).getID()
      /home/denis/git/buildkit/solver/cachemanager.go:287 +0x1a4
  github.com/moby/buildkit/solver.(*cacheManager).Query()
      /home/denis/git/buildkit/solver/cachemanager.go:74 +0x484
  github.com/moby/buildkit/solver/llbsolver.(*lazyCacheManager).Query()
      /home/denis/git/buildkit/solver/llbsolver/bridge.go:330 +0x12f
  github.com/moby/buildkit/solver.(*combinedCacheManager).Query.func1.1()
      /home/denis/git/buildkit/solver/combinedcache.go:43 +0xf2
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /home/denis/git/buildkit/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x82

Previous read at 0x00c000b457d0 by goroutine 32766:
  runtime.mapiternext()
      /usr/lib/go/src/runtime/map.go:866 +0x0
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:201 +0x1164
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver.(*exporter).ExportTo()
      /home/denis/git/buildkit/solver/exporter.go:172 +0x1649
  github.com/moby/buildkit/solver/llbsolver.NewProvenanceCreator.func1()
      /home/denis/git/buildkit/solver/llbsolver/provenance.go:448 +0x310
  github.com/moby/buildkit/solver/llbsolver.(*ProvenanceCreator).Predicate()
      /home/denis/git/buildkit/solver/llbsolver/provenance.go:493 +0x161
  github.com/moby/buildkit/solver/llbsolver.(*Solver).recordBuildHistory.func1.1()
      /home/denis/git/buildkit/solver/llbsolver/solver.go:205 +0x10d
  github.com/moby/buildkit/solver/llbsolver.(*Solver).recordBuildHistory.func1.2()
      /home/denis/git/buildkit/solver/llbsolver/solver.go:243 +0x108
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /home/denis/git/buildkit/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x82

Goroutine 32771 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /home/denis/git/buildkit/vendor/golang.org/x/sync/errgroup/errgroup.go:72 +0x12e
  github.com/moby/buildkit/solver.(*combinedCacheManager).Query.func1()
      /home/denis/git/buildkit/solver/combinedcache.go:42 +0x273
  github.com/moby/buildkit/solver.(*combinedCacheManager).Query()
      /home/denis/git/buildkit/solver/combinedcache.go:56 +0x274
  github.com/moby/buildkit/solver.(*cacheWithCacheOpts).Query()
      <autogenerated>:1 +0xc8
  github.com/moby/buildkit/solver.(*edge).probeCache()
      /home/denis/git/buildkit/solver/edge.go:198 +0x1a4
  github.com/moby/buildkit/solver.(*edge).processUpdate()
      /home/denis/git/buildkit/solver/edge.go:429 +0x1927
  github.com/moby/buildkit/solver.(*edge).unpark()
      /home/denis/git/buildkit/solver/edge.go:329 +0xd9
  github.com/moby/buildkit/solver.(*scheduler).dispatch()
      /home/denis/git/buildkit/solver/scheduler.go:136 +0x7b0
  github.com/moby/buildkit/solver.(*scheduler).loop()
      /home/denis/git/buildkit/solver/scheduler.go:104 +0x30a
  github.com/moby/buildkit/solver.newScheduler.func1()
      /home/denis/git/buildkit/solver/scheduler.go:35 +0x39

Goroutine 32766 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /home/denis/git/buildkit/vendor/golang.org/x/sync/errgroup/errgroup.go:72 +0x12e
  github.com/moby/buildkit/solver/llbsolver.(*Solver).recordBuildHistory.func1()
      /home/denis/git/buildkit/solver/llbsolver/solver.go:242 +0xa0d
  github.com/moby/buildkit/solver/llbsolver.(*Solver).Solve.func2()
      /home/denis/git/buildkit/solver/llbsolver/solver.go:460 +0xaf
  runtime.deferreturn()
      /usr/lib/go/src/runtime/panic.go:476 +0x32
  github.com/moby/buildkit/control.(*Controller).Solve()
      /home/denis/git/buildkit/control/control.go:426 +0x1b1a
  github.com/moby/buildkit/api/services/control._Control_Solve_Handler.func1()
      /home/denis/git/buildkit/api/services/control/control.pb.go:2438 +0x88
  github.com/moby/buildkit/util/grpcerrors.UnaryServerInterceptor()
      /home/denis/git/buildkit/util/grpcerrors/intercept.go:14 +0x7c
  github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1()
      /home/denis/git/buildkit/vendor/github.com/grpc-ecosystem/go-grpc-middleware/chain.go:25 +0x8e
  go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1()
      /home/denis/git/buildkit/vendor/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go:342 +0x6fa
  main.unaryInterceptor.func1()
      /home/denis/git/buildkit/cmd/buildkitd/main.go:593 +0x27c
  github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1()
      /home/denis/git/buildkit/vendor/github.com/grpc-ecosystem/go-grpc-middleware/chain.go:25 +0x8e
  github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1()
      /home/denis/git/buildkit/vendor/github.com/grpc-ecosystem/go-grpc-middleware/chain.go:34 +0x126
  github.com/moby/buildkit/api/services/control._Control_Solve_Handler()
      /home/denis/git/buildkit/api/services/control/control.pb.go:2440 +0x1dd
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /home/denis/git/buildkit/vendor/google.golang.org/grpc/server.go:1336 +0x177a
  google.golang.org/grpc.(*Server).handleStream()
      /home/denis/git/buildkit/vendor/google.golang.org/grpc/server.go:1704 +0xff8
  google.golang.org/grpc.(*Server).serveStreams.func1.2()
      /home/denis/git/buildkit/vendor/google.golang.org/grpc/server.go:965 +0xec
```

https://github.com/moby/buildkit/blob/fec5a04d16bece638ee44c1e5b728f7b978b68cb/solver/exporter.go#L201